roles/kafka_controller/tasks/health_check.yml

```
@@ -0,0 +1,44 @@
+---
+# health check for kafka controller
+- name: Add controller listener in client properties
+  shell: |
+    echo confluent.use.controller.listener=true >> {{kafka_controller.client_config_file}}
+  ignore_errors: true
+  changed_when: false
+  check_mode: false
+  when: confluent_server_enabled|bool
+
+- name: Check kafka-metadata-quorum
+  shell: |
+    {{ binary_base_path }}/bin/kafka-metadata-quorum --bootstrap-server {{inventory_hostname}}:9093 \
+      --command-config {{kafka_controller.client_config_file}} describe --replication
+  environment:
+    KAFKA_OPTS: "-Xlog:all=error -XX:+IgnoreUnrecognizedVMOptions"
+  ignore_errors: false
+  changed_when: false
+  check_mode: false
+  when: confluent_server_enabled|bool # metadata-quorum doesn't work on community kafka
+
+- name: Register LogEndOffset
+  shell: |
+    {{ binary_base_path }}/bin/kafka-metadata-quorum --bootstrap-server {{inventory_hostname}}:9093 \
+      --command-config {{kafka_controller.client_config_file}} describe --replication | awk '{print $2}'
+  environment:
+    KAFKA_OPTS: "-Xlog:all=error -XX:+IgnoreUnrecognizedVMOptions"
+  register: LEO
+  ignore_errors: false
+  changed_when: false
+  check_mode: false
+  when: confluent_server_enabled|bool
```

```suggestion
- name: Register LogEndOffset
  shell: |
    {{ binary_base_path }}/bin/kafka-metadata-quorum --bootstrap-server {{inventory_hostname}}:9093 \
      --command-config {{kafka_controller.client_config_file}} describe --replication | awk '{print $2}'
  environment:
    KAFKA_OPTS: "-Xlog:all=error -XX:+IgnoreUnrecognizedVMOptions"
  register: LEO
  ignore_errors: false
  changed_when: false
  check_mode: false
  when: confluent_server_enabled|bool
```

looks like duplicate code

---

so the reason we added 2 tasks was the presence of  " | awk '{print $2}'" in the task Register LogEndOffset. Due to this LEO always registers some value and was not failing even though the metadata-quorum command failed.

---

Aw interesting! I'd put that in a comment then for readability. Also AFAIK, you need to filter out the observers from the output of that command. So if you get:
```
$ kafka-metadata-quorum --bootstrap-server localhost:9074 --command-config /tmp/command.properties describe --replication
NodeId	LogEndOffset	Lag	LastFetchTimestamp	LastCaughtUpTimestamp	Status
9991  	307962      	0  	1673549949902     	1673549949902        	Leader
9990  	307962      	0  	1673549949890     	1673549949890        	Follower
9992  	307962      	0  	1673549949887     	1673549949887        	Follower
0     	307962      	0  	1673549949890     	1673549949890        	Observer
1     	307962      	0  	1673549949889     	1673549949889        	Observer
2     	307962      	0  	1673549949887     	1673549949887        	Observer
```

the kraft controllers are the first 3, and the brokers are the next 3. I think we only care that the LogEndOffset of the 3 3 really matter. so maybe add a `grep -v Observer`